src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs

```diff
@@ -30,17 +30,38 @@ public abstract class PageModel : IAsyncPageFilter, IPageFilter
         private IObjectModelValidator _objectValidator;
         private ITempDataDictionary _tempData;
         private IUrlHelper _urlHelper;
+        private PageContext _pageContext;
 
         /// <summary>
         /// Gets the <see cref="RazorPages.PageContext"/>.
         /// </summary>
         [PageContext]
-        public PageContext PageContext { get; set; }
+        public PageContext PageContext
+        {
+            get
+            {
+                if (_pageContext == null)
+                {
+                    _pageContext = new PageContext();
+                }
+
+                return _pageContext;
+            }
+            set
+            {
+                if (value == null)
```

Comment:
I always have mixed feelings about disallowing this. It would be kind of a neat way to "reset" the value, but it's also weird because if you *did* set `null`, then you'd never get `null` back.

Thoughts?

(I think this is fine the way it is, I just wanted to pose the question.)

Reply:
I don't like *neat* things.

We already do this in a bunch of places (lazy initialize, but throw if you try to set a null instance) and no one has complained.